modules/RouterUtils.js

```diff
@@ -1,26 +1,7 @@
-import deprecateObjectProperties from './deprecateObjectProperties'
-
 export function createRouterObject(history, transitionManager) {
   return {
     ...history,
     setRouteLeaveHook: transitionManager.listenBeforeLeavingRoute,
```

Comment:
We can't expose that `listen` as-is: https://github.com/reactjs/react-router/blob/v2.3.0/modules/createTransitionManager.js#L261-L262

I'm also a bit scared of going down this route. If we make it easy to listen to what's in the router, then inevitably someone is going to try to build a Redux integration. That integration won't work properly with Redux devtools (because router state can't be serialized), and will have the same subtle state management bugs that React Router Redux did until we rewrote it to wrap the history and own the state.